prov/rxm/src/rxm_msg.c

```diff
@@ -53,7 +53,7 @@ int rxm_handle_unexp_sar(struct rxm_recv_queue *recv_queue,
 	ret = rxm_handle_rx_buf(rx_buf);
 	last = rxm_sar_get_seg_type(&rx_buf->pkt.ctrl_hdr) == RXM_SAR_SEG_LAST;
 	if (ret || last)
-		return ret;
+		return (int)ret;
```

Comment:
The function should be returning ssize_t rather than an int.  The places where it's called return ssize_t.